arviz/plots/loopitplot.py

```diff
@@ -206,20 +206,19 @@ def plot_loo_pit(
             )
             fill_kwargs.setdefault("label", "{:.3g}% credible interval".format(credible_interval))
     else:
-        loo_pit_kde, _, _ = _fast_kde(loo_pit, xmin=0, xmax=1)
+        loo_pit_kde, xmin, xmax = _fast_kde(loo_pit)
 
         unif = np.random.uniform(size=(n_unif, loo_pit.size))
-        x_vals = np.linspace(0, 1, len(loo_pit_kde))
+        x_vals = np.linspace(xmin, xmax, len(loo_pit_kde))
         if use_hdi:
+            n_obs = len(loo_pit_kde)
```

Comment:
Why `len(loo_pit_kde)` and not `loo_pit.size`? Shouldn't the trend "the more observations the more restrictive hdi should be"?

Reply:
The asymptotic distribution should be uniform on the interval 0 and 1. If you do not have values covering that range then the model is miss-calibrated, even if it looks flat in a subinterval. I think not setting the range between 0 and 1. Makes this problem even easier to spot. 
You are right it should be loo_pit.size.

Reply:
I was also underestimating the normalization effect, loo pit kde generally ends up clearly above the upper limit of unif kdes